### ISSUE TITLE
`unreachable_code` false positive for a loop with a try operator

### ISSUE THREAD

**Issue by KAAtheWiseGit** [rust-lang/rust#134685]
### Code

```Rust
fn fallible(i: usize) -> Result<usize, ()> {
	if i < 10 {
		Ok(i + 1)
	} else {
		Err(())
	}
}

fn main() -> Result<(), ()> {
	let mut i = 0;

	loop {
		i = fallible(i)?;
	}

	println!("done");
}
```

### Current output

```Shell
warning: unreachable statement
  --> src/main.rs:16:2
   |
12 | /     loop {
13 | |         i = fallible(i)?;
14 | |     }
   | |_____- any code following this expression is unreachable
15 |
16 |       println!("done");
   |       ^^^^^^^^^^^^^^^^ unreachable statement
   |
   = note: `#[warn(unreachable_code)]` on by default
   = note: this warning originates in the macro `println` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: 1 warning emitted
```

### Desired output

```Shell

```

### Rationale and extra context

The compiler doesn't account for the try operator terminating the loop, resulting in a spurious `unreachable_code` warning.

### Other cases

```Rust

```

### Rust Version

```Shell
rustc 1.83.0 (90b35a623 2024-11-26) (Alpine Linux 1.83.0-r0)
binary: rustc
commit-hash: 90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf
commit-date: 2024-11-26
host: x86_64-alpine-linux-musl
release: 1.83.0
LLVM version: 19.1.4
```

**Comment by cyrgani** [comment]
Why is this a false positive? The only way to exit the loop is if `fallible` returns an `Err`, which will be returned by `?` immediately. The equivalent line without `?` would look like this:
```rust
i = match fallible(i) {
    Ok(o) => o,
    Err(e) => return Err(e.into()),
};
```
This should make it more obvious that the `println!` is really unreachable.

**Comment by KAAtheWiseGit** [comment]
Ah, that's correct!

I mixed up the control flow between breaking and returning, apologies.
